ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerImpl.java

```diff
@@ -509,6 +513,10 @@ public void onOptionSelected() {
             new DevOptionHandler() {
               @Override
               public void onOptionSelected() {
+                if (!mDevSettings.isHotModuleReplacementEnabled() && !mDevSettings.isJSDevModeEnabled()) {
+                  Toast.makeText(mApplicationContext, "Dev mode needs to be turned on to enable HMR.", Toast.LENGTH_LONG).show();
```

Comment:
I think the message isn't sufficiently actionable (as opposed to the one above) because of the return afterwards. Could we either make it enable dev mode or change the message to include "Enable Dev mode first to enable HMR"? I think I'd probably prefer it just enabling dev mode and HMR both, and showing the message you have here. What do you think?

Reply:
yes, I was gonna suggest this as well.

Reply:
@cpojer please let me know what you think, thanks

Reply:
also (that would be something for another PR) - I find that for me, the only reason I go to the "Dev Settings" is to turn off / on the dev mode. Perhaps it would make sense to move that option to the dialog, same as with turning on / off the HMR